vhdbuilder/packer/list-images.ps1

```diff
@@ -0,0 +1,91 @@
+<#
+    .SYNOPSIS
+        Produces a JSON image BOM for a Windows VHD
+
+    .DESCRIPTION
+        Produces a JSON image BOM for a Windows VHD
+#>
+$ErrorActionPreference = "Stop"
+
+$imageBomJsonFilePath = "c:\image-bom.json"
+$bomList = @()
+
+# starting containerd for printing containerD info, the same way as we pre-pull containerD images in configure-windows-vhd.ps1
+Start-Job -Name containerd -ScriptBlock { containerd.exe }
+$imageList=$(ctr.exe -n k8s.io image ls | select -Skip 1)
+foreach($image in $imageList) {
+    $splitResult=($image -split '\s+')
+    if ($splitResult[0].StartsWith("sha256:")) {
+        # Get repoDigests from sha256
+        # Example:
+        # sha256:1fb25eb608ab558cf66b7546fd598b2dc81b6678a380fcdb188da780e107f4ab
+        # application/vnd.docker.distribution.manifest.list.v2+json
+        # sha256:dfd3ce22e4b6e987ff2bfb3efe5e4912512fce35660be2ae5faa91e6f4da9748
+        # 1.3
+        # GiB
+        # windows/amd64
+        # io.cri-containerd.image=managed
+        $repoDigests=$splitResult[0]
+        $id=$splitResult[2]
+
+        $isExist=$false
+        foreach($bom in $bomList) {
+            if ($bom.id -eq $id) {
+                $bom.repoDigests = $repoDigests
+                $isExist=$true
+                break
+            }
+        }
+        if (-not $isExist) {
+            # This should never happen
+            # We need to handle id and repoTags in the first loop and then handle repoDigests in the second loop if this occurs
+            throw "Cannot find image id $id in bomList"
+        }
+    } else {
+        # Get id and repoTags
+        # Example:
+        # mcr.microsoft.com/windows/servercore:ltsc2022
+        # application/vnd.docker.distribution.manifest.list.v2+json
+        # sha256:dfd3ce22e4b6e987ff2bfb3efe5e4912512fce35660be2ae5faa91e6f4da9748
+        # 1.3
+        # GiB
+        # windows/amd64
+        # io.cri-containerd.image=managed
+        #
+        # mcr.microsoft.com/windows/servercore@sha256:dfd3ce22e4b6e987ff2bfb3efe5e4912512fce35660be2ae5faa91e6f4da9748
+        # application/vnd.docker.distribution.manifest.list.v2+json
+        # sha256:dfd3ce22e4b6e987ff2bfb3efe5e4912512fce35660be2ae5faa91e6f4da9748
+        # 1.3
+        # GiB
+        # windows/amd64
+        # io.cri-containerd.image=managed
+        $repoTags=$splitResult[0]
+        $id=$splitResult[2]
+
+        $isExist=$false
+        foreach($bom in $bomList) {
+            if ($bom.id -eq $id) {
+                $bom.repoTags += $repoTags
```

Comment:
we don't add this for linux I think. we only have the repo digest? I believe I checked with Gaurav and he didn't need the image id (hash of layers), only the manifest sha (which you have already).

may want to double check with Gaurav

Reply:
as ref https://github.com/Azure/AgentBaker/blob/7a0ef4ff0090e9581ccfc97e5c3b4477db1863ad/vhdbuilder/release-notes/AKSUbuntu/gen2/2204containerd/latest-image-list.json#L509-L518

Reply:
sorry -- not add it as part of repoTags. it's already there as `id`.

Reply:
Thanks for the feedback. Add the code to ignore repoTags when it contains id.